js/actions/user.js

```diff
@@ -21,14 +16,34 @@ const RECEIVE_USER_PROFILE_ACTION = 'RECEIVE_USER_PROFILE_ACTION'
 const UPDATE_USER_PROFILE_REQUEST_ACTION = 'UPDATE_USER_PROFILE_REQUEST_ACTION'
 const UPDATE_USER_PROFILE_ERROR_ACTION = 'UPDATE_USER_PROFILE_ERROR_ACTION'
 
+// Creators
+function receiveUserProfile(profile: ApiUserProfile): Action {
+  return {
+    type: RECEIVE_USER_PROFILE_ACTION,
+    profile: profile,
+    receivedAt: Date.now()
+  }
+}
+
+function updateUserProfileError(): Action {
+  return {
+    type: UPDATE_USER_PROFILE_ERROR_ACTION,
+    data: {
+      error: I18n.t('errors.profileUpdateError')
+    }
+  }
+}
+
 /**
  * Begins an API requests for the user profile to the backend.
  */
-function requestUserProfile(): ThunkAction {
+function requestUserProfile(): Thunk {
   return (dispatch: Dispatch, getState: GetState) => {
     const user = getState().user
     // If the user is not logged in we can't request the profile
-    if (!user.isLoggedIn) return
+    if (!user.isLoggedIn) {
```

Comment:
note for later: is there a way we can enforce at "compile time" for this action to be called only when the user is in the logged in state?

Reply:
All thunks are going to be replaced by sagas. Using saga we can decide start listening for PROFILE_LOAD only after after USER_LOGIN_SUCCESS  so the code will no longer require this check.

Reply:
awesome